src/value_cmp.c

```diff
@@ -19,15 +19,17 @@ int cmp_string(void *p1, void *p2) {
     return -1;
 
 
+  int v1_len = strlen(v1->stringval);
+  int v2_len = strlen(v2->stringval);
   // compare the longest length possible, which is the shortest length of the
   // two strings
-  int cmp = strncasecmp(v1->stringval.str, v2->stringval.str,
-                    MIN(v2->stringval.len, v1->stringval.len));
+  int cmp = strncasecmp(v1->stringval, v2->stringval,
```

Comment:
I'm not sure, but isn't strcmp similarly to our implementation, in case strings aren't equal in length but have the same common prefix?

Reply:
Good call - this can be massively simplified.